pkg/bgpv1/agent/controller.go

```diff
@@ -431,62 +429,51 @@ func (c *Controller) FullWithdrawal(ctx context.Context) {
+				n.PeerPort = pointer.Int32(v2alpha1api.DefaultBGPPeerPort)
 			}
-			if n.HoldTime.Duration == 0 {
+			if n.ConnectRetryTimeSeconds == nil || *n.ConnectRetryTimeSeconds == 0 {
+				// As GoBGP defaults the ConnectRetryTime for 0 value, it cannot be 0 in our case.
+				n.ConnectRetryTimeSeconds = pointer.Int32(v2alpha1api.DefaultBGPConnectRetryTimeSeconds)
+			}
+			if n.HoldTimeSeconds == nil || *n.HoldTimeSeconds == 0 {
 				// RFC4271 Sec 4.4 says that hold time can be 0 and has a special meaning that disables keepalive.
 				// However, as GoBGP defaults the hold time for 0 value, it cannot be 0 in our case.
-				n.HoldTime.Duration = types.DefaultBGPHoldTime
+				n.HoldTimeSeconds = pointer.Int32(v2alpha1api.DefaultBGPHoldTimeSeconds)
 			}
-			if n.KeepAliveTime.Duration == 0 {
-				n.KeepAliveTime.Duration = n.HoldTime.Duration / 3
+			if n.KeepAliveTimeSeconds == nil || *n.KeepAliveTimeSeconds == 0 {
+				// As GoBGP defaults the KeepAliveTime for 0 value (to HoldTime / 3), it cannot be 0 in our case.
+				n.KeepAliveTimeSeconds = pointer.Int32(v2alpha1api.DefaultBGPKeepAliveTimeSeconds)
 			}
-
 			// apply graceful restart defaults
-			if n.GracefulRestart.Enabled && n.GracefulRestart.RestartTime.Duration == 0 {
-				n.GracefulRestart.RestartTime.Duration = types.DefaultGRRestartTime
+			if n.GracefulRestart != nil && n.GracefulRestart.Enabled &&
+				(n.GracefulRestart.RestartTimeSeconds == nil || *n.GracefulRestart.RestartTimeSeconds == 0) {
+				// As GoBGP defaults the RestartTimeSeconds for 0 value (to HoldTime), it cannot be 0 in our case.
+				n.GracefulRestart.RestartTimeSeconds = pointer.Int32(v2alpha1api.DefaultBGPGRRestartTimeSeconds)
 			}
 		}
 	}
 	return p
 }
 
 // validatePolicy validates the CiliumBGPPeeringPolicy.
+// The validation is normally done by kube-apiserver (based on CRD validation markers),
+// this validates only those conditions that cannot be enforced by them.
 func (c *Controller) validatePolicy(policy *v2alpha1api.CiliumBGPPeeringPolicy) error {
 	for _, r := range policy.Spec.VirtualRouters {
 		for _, n := range r.Neighbors {
-			if n.ConnectRetryTime.Duration < 0 {
-				return fmt.Errorf("connectRetryTime is negative for peer ASN %d, IP %s", n.PeerASN, n.PeerAddress)
-			}
-			if n.HoldTime.Duration < minHoldTime {
-				// RFC4271 Sec 4.2 says that the hold time MUST be zero or at least 3 seconds.
-				// However, as GoBGP defaults the hold time for 0 value, it cannot be 0 in our case.
-				return fmt.Errorf("holdTime is lower than %v for peer ASN %d, IP %s", minHoldTime, n.PeerASN, n.PeerAddress)
-			}
-			if n.KeepAliveTime.Duration < 0 {
-				return fmt.Errorf("keepAliveTime is negative for peer ASN %d, IP %s", n.PeerASN, n.PeerAddress)
-			}
-			if n.KeepAliveTime.Duration > n.HoldTime.Duration {
+			// nil checks just in case, as everything should be defaulted in applyPolicyDefaults()
+			if n.KeepAliveTimeSeconds != nil && n.HoldTimeSeconds != nil &&
```

Comment:
I think we should add validateTimers method and move this logic into it.

Reply:
Did you mean a method of `CiliumBGPNeighbor` in the `v2alpha1` package?

Reply:
I think it would make a lot of sense, done that in https://github.com/cilium/cilium/pull/26040/commits/956f08bb95b47ccaf134d2851b3ac03949d12cd0